Mixpanel/MixpanelInstance.swift

```diff
@@ -845,19 +850,42 @@ extension MixpanelInstance {
      - parameter completion: an optional completion handler for when the flush has completed.
      */
     open func flush(completion: (() -> Void)? = nil) {
-        serialQueue.async() {
+        
+        self.archive()
+
+        networkQueue.async() {
```

Comment:
we should probably still keep the flush ordered. so we should wait for the trackingQueue to finish the tasks it set to do before the flush was invoked and only then issue the flush